src/Monitors/RockblockReportMonitor.cpp

```diff
@@ -23,47 +22,113 @@ void RockblockReportMonitor::execute()
         for (auto &data : sfr::rockblock::camera_report) {
             sfr::rockblock::downlink_report.push_back(data);
         }
-        break;
+        return;
 
     case report_type::normal_report:
         sfr::rockblock::downlink_report.clear();
         for (auto &data : sfr::rockblock::normal_report) {
             sfr::rockblock::downlink_report.push_back(data);
         }
-        break;
+        return;
 
     case report_type::imu_report:
         sfr::rockblock::downlink_report.clear();
         for (auto &data : sfr::rockblock::imu_report) {
             sfr::rockblock::downlink_report.push_back(data);
         }
-        break;
+        return;
     }
 }
 
+bool RockblockReportMonitor::check_in_low_power_mode()
+{
+    return sfr::mission::current_mode == sfr::mission::lowPowerAliveSignal ||
+           sfr::mission::current_mode == sfr::mission::lowPowerDetumbleSpin ||
+           sfr::mission::current_mode == sfr::mission::lowPower ||
+           sfr::mission::current_mode == sfr::mission::lowPowerDeployment ||
+           sfr::mission::current_mode == sfr::mission::lowPowerArmed ||
+           sfr::mission::current_mode == sfr::mission::lowPowerInSun;
+}
```

Comment:
Another way to do this is to check if mode_type is LP using get_type(). Please see https://github.com/Alpha-CubeSat/oop-flight-code/blob/main/src/Modes/mode_type.enum.